File: pset6/readability/readability.py
def main():

    text = input("Text: ")
    letters = 0
    words = 0
    in_word = 0
    sentences = 0

    for char in text:
        if char.isalpha():
            if in_word == 0:
                in_word = 1
                words += 1
            letters += 1
        elif char in ['.', '!', '?']:
            sentences += 1
        elif char == ' ':
            in_word = 0
        
    cs_coleman_liau(letters, words, sentences)

def cs_coleman_liau(letters, words, sentences):
    l = (letters / words) * 100
    s = (sentences / words) * 100
    index = 0.0588 * l - 0.296 * s - 15.8

    rank = cs_round(index)

    if rank > 16:
        print("Grade 16+")
    elif rank < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {rank}")

def cs_round(nbr):
    interger = int(nbr)
    decimal = float(nbr - interger)

    if decimal >= 0.5:
        return interger + 1
    else:
        return interger

main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>

void ms_coleman_liau(int letters, int words, int sentence);
int ms_round(float nbr);
int ms_strlen(string str);

int main(void)
{
    string text = get_string("Text: ");
    int i = 0;
    int letters = 0;
    int words = 0;
    int in_word = 0;
    int sentences = 0;

    while (text[i] != '\0')
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            if (in_word == 0)
            {
                in_word = 1;
                words++;
            }
            letters++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            sentences++;
        else if (text[i] == ' ')
            in_word = 0;
        i++;
    }
    ms_coleman_liau(letters, words, sentences);
    return (0);
}

void ms_coleman_liau(int letters, int words, int sentences)
{
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;
    int rank = ms_round(index);

    if (rank > 16)
        printf("Grade 16+\n");
    else if (rank < 1)
        printf("Before Grade 1\n");
    else
        printf("Grade %d\n", rank);
}

int ms_round(float nbr)
{
    int interger = (int) nbr;
    float decimal = nbr - interger;

    if (decimal >= 0.5)
        return (interger + 1);
    else
        return (interger);
}
